sim/core/item_swaps.go

```diff
@@ -98,20 +100,32 @@ func (swap *ItemSwap) initialize(character *Character) {
 	swap.character = character
 }
 
-func (character *Character) RegisterItemSwapCallback(slots []proto.ItemSlot, callback OnItemSwap) {
+func (character *Character) RegisterItemSwapCallbackForSlots(slots []proto.ItemSlot, callback OnItemSwapSlot) {
 	if character == nil || !character.ItemSwap.IsEnabled() || len(slots) == 0 {
 		return
 	}
 
 	if (character.Env != nil) && character.Env.IsFinalized() {
-		panic("Tried to add a new item swap callback in a finalized environment!")
+		panic("Tried to add a new item swap callback for slots in a finalized environment!")
 	}
 
 	for _, slot := range slots {
-		character.ItemSwap.onSwapCallbacks[slot] = append(character.ItemSwap.onSwapCallbacks[slot], callback)
+		character.ItemSwap.onSwapSlotCallbacks[slot] = append(character.ItemSwap.onSwapSlotCallbacks[slot], callback)
 	}
 }
 
+func (character *Character) RegisterItemSwapCallback(callback OnItemSwap) {
```

Comment:
I would call this one `RegisterEquipSwapCallback()`, and keep the other one named `RegisterItemSwapCallback()` like before.

Reply:
Wait, the only use of this across the codebase is within the `UpdateSubactions()` code that you don't even need, so this extra callback structure you added can just be deleted I think?

Reply:
Yep! Already removed it before I ended at this comment 😂